pytorchfi/core.py

```diff
@@ -66,6 +67,28 @@ def __init__(self, model, h, w, batch_size, **kwargs):
             )
         )
 
+    def _traverseModelAndSetHooks(self, model, layer_types):
+        handles = []
+        shape = []
+        for layer in model.children():
+            # leaf node
+            if list(layer.children()) == []:
+                for i in layer_types:
+                    if isinstance(layer, i):
```

Comment:
Will this append the hook to every instance of the provided layer? Or is there an option where we're able to just inject into a specific layer (for example, the first)

Reply:
At this stage, it appends the hook to all instances of `layer_type[i]`

During injection, the user can specify which specific layer of interest to perform the injection.

tl;dr: the hook is appended to all the layers. The injection location is specified later on by the user.